web/packages/build/package.json

```diff
@@ -69,15 +69,15 @@
     "jsdom-testing-mocks": "^1.13.0",
     "msw": "^0.47.4",
     "node-gyp": "^10.1.0",
-    "react": "^18.2.0",
-    "react-dom": "^18.2.0",
+    "react": "^18.3.1",
```

Comment:
The failures related to `findDOMNode` and `defaultProps` should not cause the build to fail. I'll take care of this in a sec.

Reply:
The warnings cause the tests to fail (`jest-fail-on-console`).

Reply:
The purpose of jest-fail-on-console was to catch unexpected warnings in the code that we ourselves write – it was mostly about the `act` warnings which are hard to catch otherwise.

The purpose of warnings related to deprecation is to let people know that something will have to be updated before the next major release. I don't think we have to migrate from deprecated APIs the moment they become deprecated.

Without changing anything, we get 167 failures. After taking care of `defaultProps` from `Button`, `ButtonLink` and `LogoHero`, we have 74 failures. The next big thing to address would be react-select update and our own components that use `defaultProps`.

This change results in one snapshot update being required in teleport.e.